bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplacer.java

```diff
@@ -0,0 +1,778 @@
+package org.eclipse.ui.texteditor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.PatternSyntaxException;
+
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Display;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+import org.eclipse.jface.text.IFindReplaceTargetExtension;
+import org.eclipse.jface.text.IFindReplaceTargetExtension3;
+import org.eclipse.jface.text.IFindReplaceTargetExtension4;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.Region;
+
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.texteditor.NLSUtility;
+
+/**
+ * @since 3.18
+ */
+public class FindReplacer {
```

Comment:
Could it be made only package visible at the moment?

Reply:
As an alternative, it may be moved to an internal package? Actually, it could be moved to some more generic package than one related to `texteditor`, as this functionality will not be restricted to text editors according to the proposal #1090 (not necessarily now, but in the future, so it should be made part of the public API right now).